### DISCUSSION TITLE
Browsersync not working again on Livewire

### DISCUSSION THREAD

**Discussion by bzarzuela** [livewire/livewire#4835] [1. Help] [answered]
It seems like I have the same problem that was fixed in [#2481](https://github.com/livewire/livewire/pull/2481).

Here's the error, shared on Flare: https://flareapp.io/share/Bm0MaNeP#F50

I'm running Livewire v2.10.4.

I have a pretty simple component that shows a Sign In and Register button that I use on my Blade templates to give the user an easy way to go back to where they were after signing in or registering.

```
@guest
    <livewire:sign-in-or-register url="{{ url()->full() }}" />
@endguest
```

```
<div class="flex justify-content-between space-x-4">
    <form wire:submit.prevent="loginThenGoBack()" class="w-full">
        <button type="submit" class="mt-8 w-full bg-indigo-600 border border-transparent rounded-md py-3 px-8 flex items-center justify-center text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500">
            Sign In
        </button>
    </form>

    <form wire:submit.prevent="registerThenGoBack()" class="w-full">
        <button type="submit" class="mt-8 w-full bg-indigo-600 border border-transparent rounded-md py-3 px-8 flex items-center justify-center text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500">
            Register
        </button>
    </form>
</div>
```

```
<?php

namespace App\Http\Livewire;

use Illuminate\Contracts\View\View;
use Livewire\Component;

class SignInOrRegister extends Component
{
    public string $url;

    public function render(): View
    {
        return view('livewire.sign-in-or-register');
    }

    /** @noinspection PhpMissingReturnTypeInspection */
    public function loginThenGoBack()
    {
        return redirect()
            ->setIntendedUrl($this->url)
            ->to(route('login'));
    }

    /** @noinspection PhpMissingReturnTypeInspection */
    public function registerThenGoBack()
    {
        return redirect()
            ->setIntendedUrl($this->url)
            ->to(route('register'));
    }
}
```

This works properly if I access the site on its local URL being served by Valet. However, when I try to use this while on Browsersync (localhost:3000), it gives me the corrupted payload exception:

`Livewire encountered corrupt data when trying to hydrate the [sign-in-or-register] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.`

I'm not sure what I'm doing wrong, I hope you guys can point it out.

**Comment by smoglica** [comment, marked as answer]
I'm experiencing the same issue. I did different tests and apparently this issue is caused by absolute URLs.

In `webpack.mix.js` file I have this:

```js
mix.browserSync({ open: false, proxy: process.env.APP_URL });
```

Then in the livewire component I tested with:


```php
<?php
route('home'); // Returns absolute path: http://localhost:3000/
// -> Generates the error

route('home', [], false); // Returns a relative path: /
// -> Doesn't generate the error
```

Seems that hashed values doesn't match because of `http://localhost` (proxied URL) and `http://localhost:3000` (Browsersync URL) difference in the base URL.

I hope this helps to solve the problem.
